### PR TITLE
Fix docs breakage

### PR DESCRIPTION
**Pull request by bryevdv**
This PR fixes the docs breakage that was uncovered by the fix in #230 

TLDR; The #230 bug caused all the *unimplemented* functions to be erroneously cloned as a tuple `(func,)` instead of `func`. This in turn caused the `autosummary` directive for `ndarray` to ignore them. Howevever *after* #230, the `autosummary` directive for `ndarray` picked the up, and wanted to generate references for them in the class TOC. These are *broken* references, because we do not document *unimplemented* (i.e. numpy-only) functions in our docs. 

This PR removes the `autosummary` directive for `ndarray` and replaces it with a hand-curated TOC. I suppose we could dispense with the class TOC altogether, but Numpy does have a class TOC ([ref](https://numpy.org/doc/1.22/reference/generated/numpy.ndarray.html#numpy.ndarr)), and I do think they are handy. 

cc @magnatelee @ipdemes

### CODE REVIEW

File: cunumeric/coverage.py [hunk 1]
```diff
@@ -129,10 +133,12 @@ def clone_module(
...
 
-    if runtime.report_coverage:
-        for attr, value in new_globals.items():
-            if isinstance(value, FunctionType):
+    for attr, value in new_globals.items():
+        if isinstance(value, FunctionType):
+            if runtime.report_coverage:
                 new_globals[attr] = implemented(value, module_name)
+            else:
+                value._cunumeric_implemented = True
```
Comment (bryevdv) [inline cunumeric/coverage.py:141]: I would actually like to propose calling the `implemented` function unconditionally so that setting this attr can be handled in one place, but I will leave that for the future.

**Comment by bryevdv** [comment]
FYI I reset the base to `22.03` but let me know if that is not correct.

File: cunumeric/coverage.py (hunk 1 above)
Comment (magnatelee) [reply cunumeric/coverage.py:141]: yeah I also thought about doing that, and I feel we should do it now. let's make a follow-up PR for it.

**Comment by magnatelee** [comment]
Looks good to me, and it looks like this PR supersedes #237, right? If so I suggest we close #237.

**Comment by ipdemes** [comment]
@magnatelee Yes, I think this PR supersedes #237, will close it
